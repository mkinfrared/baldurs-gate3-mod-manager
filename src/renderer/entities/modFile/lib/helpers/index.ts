export { extractPakFile } from "./extractPakFile";

export { getModFiles } from "./getModFiles";

export { getModInfo } from "./getModInfo";

export { readZipInfo } from "./readZipInfo";
