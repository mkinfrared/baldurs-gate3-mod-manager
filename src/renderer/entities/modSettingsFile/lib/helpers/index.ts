export { activateMod } from "./activateMod";

export { deactivateMod } from "./deactivateMod";

export { getCurrentSettings } from "./getCurrentSettings";

export { getInstalledMods } from "./getInstalledMods";

export { getModInfo } from "./getModInfo";

export { getNodeValue } from "./getNodeValue";
