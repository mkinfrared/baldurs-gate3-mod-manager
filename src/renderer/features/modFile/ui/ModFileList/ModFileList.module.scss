.ModFileList {
  display: flex;
  flex-direction: column;
  min-height: 75dvh;

  &.isLoading {
    cursor: progress;
  }

  .header {
    font-weight: 500;

    *:last-child {
      pointer-events: none;
      visibility: hidden;
    }
  }
}
