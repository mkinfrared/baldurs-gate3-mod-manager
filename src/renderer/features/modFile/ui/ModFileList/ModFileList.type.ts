export type ModFileListProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
