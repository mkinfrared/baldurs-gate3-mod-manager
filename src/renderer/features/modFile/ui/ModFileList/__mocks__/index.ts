export { default } from "./ModFileList.mock";
