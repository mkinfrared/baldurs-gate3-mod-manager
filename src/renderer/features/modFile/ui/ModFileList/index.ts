export { ModFileList } from "./ModFileList";

export * from "./ModFileList.type";
