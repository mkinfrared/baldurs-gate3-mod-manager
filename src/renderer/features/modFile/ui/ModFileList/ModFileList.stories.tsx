import { Meta, StoryObj } from "@storybook/react";

import { ModFileList } from "./ModFileList";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "ENTITIES/ModFileList",
  component: ModFileList,
  tags: ["autodocs"],
} satisfies Meta<typeof ModFileList>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <ModFileList {...args} />
    </div>
  ),
};

export { Default };

export default meta;
