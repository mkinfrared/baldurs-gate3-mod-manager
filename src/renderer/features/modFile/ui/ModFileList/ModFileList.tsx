import { DragEventHandler, useState } from "react";

import { classNames, trpc } from "renderer/shared/lib/helpers";

import { ModFileRow } from "../ModFileRow";

import css from "./ModFileList.module.scss";
import { ModFileListProps, ReadModResult } from "./ModFileList.type";

const ModFileList = ({ className }: ModFileListProps) => {
  const [zipFiles, setZipFiles] = useState<ReadModResult>([]);
  const readFilesMutation = trpc.mod.readMods.useMutation();

  const handleDragOver: DragEventHandler = (event) => {
    event.stopPropagation();

    event.preventDefault();
  };

  const handleDrop: DragEventHandler = async (event) => {
    try {
      event.stopPropagation();

      event.preventDefault();

      const { files } = event.dataTransfer;
      const filePaths = [...files].map(({ path }) => path);
      const result = await readFilesMutation.mutateAsync(filePaths);

      setZipFiles(result);
    } catch (e) {
      console.error(e);
    }
  };

  return (
    <div
      className={classNames(css.ModFileList, className)}
      data-testid="ModFileList"
      onDragOver={handleDragOver}
      onDrop={handleDrop}
    >
      {readFilesMutation.isLoading
        ? "Loading..."
        : zipFiles?.map(({ filePath, info }, index) => (
            <ModFileRow
              key={info?.uuid ?? index}
              modFilePath={filePath}
              modName={info?.name}
              modVersion={info?.version}
            />
          ))}
    </div>
  );
};

export { ModFileList };
