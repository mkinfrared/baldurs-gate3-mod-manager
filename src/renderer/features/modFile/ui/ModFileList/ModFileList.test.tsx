import { render } from "@testing-library/react";

import { withQueryProvider } from "renderer/shared/lib/helpers/testUtils";
import { trpc } from "renderer/shared/lib/helpers/trpc";

import { ModFileList } from "./ModFileList";

jest.mock("renderer/shared/lib/helpers/trpc", () => ({
  trpc: {
    mod: {
      readMods: {
        useMutation: jest.fn(),
      },
    },
  },
}));

describe("<ModFileList />", () => {
  const readModsMock = {
    mutateAsync: jest.fn().mockResolvedValue([]),
    isLoading: false,
  };

  const trpcMock = trpc as jest.MockedObjectDeep<typeof trpc>;

  beforeEach(() => {
    jest.clearAllMocks();

    trpcMock.mod.readMods.useMutation = jest.fn();

    trpcMock.mod.readMods.useMutation.mockImplementation(
      () => readModsMock as any,
    );
  });

  const Component = <ModFileList />;

  it("should be defined", () => {
    expect(ModFileList).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component, { wrapper: withQueryProvider });

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("ModFileList");

    expect(element).toBeDefined();
  });
});
