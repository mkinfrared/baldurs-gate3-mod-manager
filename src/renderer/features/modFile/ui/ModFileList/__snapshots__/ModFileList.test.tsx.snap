// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModFileList /> should match the snapshot 1`] = `
<div>
  <div
    class="ModFileList"
    data-testid="ModFileList"
  >
    <div
      class="Card ModFileRow header"
      data-testid="Card"
    >
      <span>
        Modification Name
      </span>
      <span>
        Version
      </span>
      <button
        class="Button ModFileButton contained primary"
        data-testid="ModFileButton"
        type="button"
      >
        <svg
          class="Spinner spinner"
        >
          bars.svg
        </svg>
        <span
          class="content"
        >
          Install
        </span>
      </button>
    </div>
    <div
      class="Card EmptyList"
      data-testid="Card"
    >
      <h4
        class="Heading header"
        data-testid="Heading"
      >
        Drag and drop you zip files here or
      </h4>
      <button
        class="Button contained primary"
        data-testid="Button"
        type="button"
      >
        <svg
          class="Spinner spinner"
        >
          bars.svg
        </svg>
        <span
          class="content"
        >
          <label
            for="file"
          >
            Click this button to select files
          </label>
        </span>
      </button>
      <input
        accept="zip"
        id="file"
        multiple=""
        type="file"
      />
    </div>
  </div>
</div>
`;
